apis/v1alpha2/httproute_types.go

```diff
@@ -336,18 +319,38 @@ type HTTPHeaderMatch struct {
 	// +kubebuilder:default=Exact
 	Type *HeaderMatchType `json:"type,omitempty"`
 
-	// Values is a map of HTTP Headers to be matched.
-	// It MUST contain at least one entry.
+	// Name is the name of the HTTP Header to be matched. Name matching MUST be
+	// case insensitive. (See
+	// https://tools.ietf.org/html/rfc7230#section-3.2).
 	//
-	// The HTTP header field name to match is the map key, and the
-	// value of the HTTP header is the map value. HTTP header field name matching
-	// MUST be case-insensitive.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	Name string `json:"Name"`
```

Comment:
I don't feel strongly about this but is `Key` a better name here?
In my interpretation, API conventions don't explicitly call out that this field should be `Name`.

Reply:
Yeah I went back and forth between the two. Ultimately I chose "name" due to the RFC using "header name": https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Admittedly the same argument could be used for "key" for query params, but I'd rather keep the 2 structures consistent in this API: https://datatracker.ietf.org/doc/html/rfc3986#section-3.4.

Reply:
A quick search of the k8s codebase shows both `key` and `name` are used in these cases, though it looks like `name` is a bit more common. As an example, this pattern is used for Pod DNS config: https://github.com/kubernetes/kubernetes/blob/fbdc4f6acbc881eb26544d9d68ae520b7cbad3e7/pkg/apis/core/types.go#L3055

Reply:
I don't feel strongly about it. Let's leave this as `name`. This thread will help in future if it comes up. Resolving now.